libs/deer/tests/test_impls_core_num.rs

```diff
@@ -106,3 +116,45 @@ fn usize_err_zero() {
         &[Token::USize(0)],
     );
 }
+
+#[cfg(not(miri))]
+proptest! {
+    #[test]
+    fn wrapping_ok(value in any::<u8>()) {
+        let expected = Wrapping(value);
+
+        assert_tokens(&expected, &[Token::Number(Number::from(value))]);
+    }
+
+    #[test]
+    fn saturating_ok(value in any::<u8>()) {
+        let expected = Saturating(value);
+
+        assert_tokens(&expected, &[Token::Number(Number::from(value))]);
+    }
+}
```

Comment:
Could we add some more tests for larger numbers like `usize`, `u64`, and `u128`? Also, it would be good if we do the same for `Wrapping` and `Saturating`.

Reply:
I can also add them to `Wrapping` and `Saturating`, but because they are simply `T::deserialize.map(Wrapping)` I thought it was a bit unnecessary if we would go another route (as proposed in the open questions), I think separate tests to look at saturating and wrapping behavior will be necessary.

Integral values are tested in https://github.com/hashintel/hash/blob/7633fdd6be8501adf4023dcea732a73ecff3abde/libs/deer/tests/test_impls_core_integral.rs